bundle/Tab/LocationView/UrlsTab.php

```diff
@@ -0,0 +1,108 @@
+use Ibexa\Contracts\Core\Repository\Values\Content\Location;
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Twig\Environment;
+
+use function array_search;
+use function preg_match;
+
+final class UrlsTab extends AbstractEventDispatchingTab implements OrderedTabInterface
+{
+    private const SYSTEM_URL_REGEX_PATTERN = '#/view/content/\d+/full/\d+/\d+$#';
+
+    public function __construct(
+        private readonly IbexaUrlsTab $inner,
+        private readonly RouterInterface $router,
+        private readonly ConfigResolverInterface $configResolver,
+        private readonly array $siteaccessList,
+        private readonly bool $showExternalSiteaccessUrls,
+        Environment $twig,
+        TranslatorInterface $translator,
+        EventDispatcherInterface $eventDispatcher,
+    ) {
+        parent::__construct($twig, $translator, $eventDispatcher);
+    }
+
+    public function getIdentifier(): string
+    {
+        return $this->inner->getIdentifier();
+    }
+
+    public function getName(): string
+    {
+        return $this->inner->getName();
+    }
+
+    public function getOrder(): int
+    {
+        return $this->inner->getOrder();
+    }
+
+    public function getTemplate(): string
+    {
+        return $this->inner->getTemplate();
+    }
+
+    public function getTemplateParameters(array $contextParameters = []): array
+    {
+        $contentTreeUrls = [];
+        $externalUrls = [];
+
+        /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Location $location */
+        $location = $contextParameters['location'];
+
+        $locationPath = $location->path;
+        foreach ($this->siteaccessList as $siteaccess) {
+            $url = $this->router->generate(
+                'ibexa.url.alias',
+                [
+                    'locationId' => $location->id,
+                    'siteaccess' => $siteaccess,
+                ],
+                UrlGeneratorInterface::ABSOLUTE_URL,
+            );
+
+            // checks if the url matches invalid system url format
+            if (preg_match(self::SYSTEM_URL_REGEX_PATTERN, $url)) {
+                continue;
+            }
+
+            $locationIdIndex = array_search((string) $location->id, $locationPath, true);
+            $rootLocationId = $this->configResolver->getParameter(
+                'content.tree_root.location_id',
+                null,
+                $siteaccess,
+            );
+            $rootLocationIdIndex = array_search((string) $rootLocationId, $locationPath, true);
+
+            // checks if the url is inside configured siteaccess content tree
+            if ($locationIdIndex !== false && $rootLocationIdIndex !== false && $rootLocationIdIndex <= $locationIdIndex) {
```

Comment:
If the `$locationIdIndex` is always index of the last element in the `$location->path`, is it necessary to compare it with `$rootLocationIdIndex`? It seems to me `$rootLocationIdIndex <= $locationIdInde` will always evaluate as `true`.

Also, please extract the check into a method:

```php
private function isUnderConfiguredContentTreeRoot(Location $location, string $siteaccess): bool
{
    $rootLocationId = $this->configResolver->getParameter(
        'content.tree_root.location_id',
         null,
         $siteaccess,
    );

    return in_array((string) $rootLocationId, $location->path, true);
}
```

Then you can get rid of the inline comment.

Reply:
You're right, check is totally redundant, fixed in https://github.com/netgen/ibexa-admin-ui-extra/pull/2/commits/122f36ae539cf712acf7a6013ccc8a7673edf842